src/Compilers/CSharp/Portable/Symbols/Source/SourceLocalSymbol.cs

```diff
@@ -685,19 +654,18 @@ protected override TypeSymbol InferTypeOfVarVariable(DiagnosticBag diagnostics)
                 {
                     case SyntaxKind.DeconstructionDeclarationStatement:
                         var localDecl = (DeconstructionDeclarationStatementSyntax)_deconstruction;
-                        var localBinder = _enclosingBinderOpt ?? this.ScopeBinder;
-                        localBinder.BindDeconstructionDeclaration(localDecl, localDecl.Assignment.VariableComponent, localDecl.Assignment.Value, diagnostics);
+                        _nodeBinder.BindDeconstructionDeclaration(localDecl, localDecl.Assignment.VariableComponent, localDecl.Assignment.Value, diagnostics);
                         break;
 
                     case SyntaxKind.ForStatement:
                         var forStatement = (ForStatementSyntax)_deconstruction;
-                        var forBinder = this.ScopeBinder.GetBinder(forStatement);
-                        forBinder.BindDeconstructionDeclaration(forStatement, forStatement.Deconstruction.VariableComponent, forStatement.Deconstruction.Value, diagnostics);
+                        Debug.Assert(this.ScopeBinder.GetBinder(forStatement) == _nodeBinder);
```

Comment:
Good question. Since you don't know, I'll open an issue about test coverage for this case.

---
In reply to: [76854336](https://github.com/dotnet/roslyn/pull/13274#discussion_r76854336) [](ancestors = 76854336,76852705,76849261)